scripts/scripts.js

```diff
@@ -0,0 +1,84 @@
+document.addEventListener('DOMContentLoaded', async () => {
+    const productInfoElement = document.querySelector('#product');
+    const productImageElement = document.querySelector('.product-image');
+    const productTitleElement = document.querySelector('.product-title');
+    const productCategoryElement = document.querySelector('.product-category');
+    const productDescriptionElement = document.querySelector('.product-description');
+    const productPriceElement = document.querySelector('.product-price');
+    const productsElement = document.querySelector('#products');
+    let data;
+    
+    const createCardElement = (card) => {
+        const cardElement = createDOMElement('div', 'card');
+        const cardImageElement = createDOMElement('img', 'card-image');
+        const cardInfoElement = createDOMElement('div', 'card-info');
+        const cardNameElement = createDOMElement('a', 'card-name');
+        const cardPriceElement = createDOMElement('div', 'card-price');
+        const cardButtonElement = createDOMElement('button', 'btn', 'btn-white', 'card-button');
+
+        !card.images || !card.images[0] ? card.images = ['images/placeholder.png'] : card.images;
+        !card.title ? card.title = "New Product" : card.title;
+        !card.category.name ? card.category.name = "No category" : card.category.name;
+        !card.description ? card.description = "No desription for this product." : card.description;
+
+        cardImageElement.setAttribute("src", card.images[0]);
```

Comment:
To handle the error of a non-existent image by the url, add this line 
```suggestion
        cardImageElement.setAttribute("src", card.images[0]);
        cardImageElement.addEventListener('error', function(e) {
            this.setAttribute("src", 'images/placeholder.png');
        })
```